mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -455,7 +508,19 @@ private[ml] object GBTClassifierParams {
     Array("logistic").map(_.toLowerCase(Locale.ROOT))
 }
 
-private[ml] trait GBTClassifierParams extends GBTParams with HasVarianceImpurity {
+private[ml] trait GBTClassifierParams extends GBTParams with HasVarianceImpurity
+  with ProbabilisticClassifierParams {
+
+  override protected def validateAndTransformSchema(
```

Comment:
in general, in this PR there are many code parts which are copy-pasted...can we dedup the code?

Reply:
Good point, I will look into it.

Reply:
@mgaido91 I tend to keep current way, that is because the superclasses are different:
1,the `super.validateAndTransformSchema(schema, fitting, featuresDataType)` in `GBTClassifierParams` & `RandomForestClassifierParams` are from `ProbabilisticClassifierParams`, which check cols probabilityCol,rawPredictionCol,featuresCol,labelCol,weightCol,predictionCol
2,while the super method called in `RandomForestRegressorParams` & `GBTRegressorParams` are from `PredictorParams`, which only check cols featuresCol,labelCol,weightCol,predictionCol

We can add another two trait for classification and regression, respectively. Like `TreeEnsembleClassifierParams` & `TreeEnsembleRegressorParams`.
However, I think this maybe not worthwhile, since there will be only two subclasses for each, and this will make the hierarchy more complex.